mooncake-store/src/storage_backend.cpp

```diff
@@ -1975,6 +1980,489 @@ tl::expected<bool, ErrorCode> BucketStorageBackend::HasNext() {
     return next_bucket_ != 0;
 }
 
+// ============================================================================
+// OffsetAllocatorStorageBackend Implementation
+// ============================================================================
+
+OffsetAllocatorStorageBackend::OffsetAllocatorStorageBackend(
+    const FileStorageConfig& file_storage_config_)
+    : StorageBackendInterface(file_storage_config_),
+      storage_path_(file_storage_config_.storage_filepath) {
+    // Cap at 90% of total_size_limit to avoid filling disk completely
+    constexpr double kDefaultQuotaPercentage = 0.9;
```

Comment:
The parameter passed in here represents the SSD quota allocated for Mooncake, not the total disk capacity, so there’s no need to worry about the disk being filled up.